### PR TITLE
Fix issues around splits not updating

### PR DESCRIPTION
**Pull request by aburgel**
/domain @Betterment/test_track_core @HipsterBrown 
/no-platform

I noticed that the chrome extension doesn't reliably update assignments, you often have to try a few times to get an assignment to "stick". I discovered two underlying issues.

1. When hitting "Save", the extension would fire an update request for all splits, regardless of whether the assignment changed, this meant that many requests were fired.

2. The extension is supposed to wait until all the update requests complete before reloading the page, but when we switched from jquery to axios the type of Promise changed from a jquery promise to a native promise. I don't think the promise waiting code works with native promises, so there would be no waiting before the page reloaded.

So combining the two issues, lots of requests and no waiting meant that some requests were canceled so those assignments were never updated.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ module.exports = function(grunt) {
 
     grunt.registerTask('hackmanifest', function() {
         // preprocessor doesn't work on .json files
-        fs.rename('dist/unpacked_extension/manifest.js', 'dist/unpacked_extension/manifest.json');
+        fs.renameSync('dist/unpacked_extension/manifest.js', 'dist/unpacked_extension/manifest.json');
```
Comment (aburgel) [inline Gruntfile.js:111]: this is needed for the newer version of grunt

File: bower.json [hunk 2]
```diff
@@ -1,18 +0,0 @@
-{
```
Comment (aburgel) [inline bower.json:1]: removing bower, npm can handle what we need.

File: src/eventBus.js [hunk 3]
```diff
@@ -14,8 +14,7 @@
             return split && TestTrack._crx.persistAssignment(split.name, split.variant, 'crx', sharedSecret);
         });
 
-        // TODO: find a way to remove usage of jQuery
-        $.when.apply($, promises).done(function() {
+        Promise.allSettled(promises).then(function() {
```
Comment (aburgel) [inline src/eventBus.js:17]: `TestTrack._crx.persistAssignment` will return a promise from axios, so now using `Promise.allSettled` should wait until all the requests complete before reloading the page

File: src/popup.js [hunk 4]
```diff
@@ -80,11 +80,11 @@ $(function() {
         $('#form').hide();
 
         var splitsToSave = $('#splits select').map(function() {
-            if ($(this).val()) {
+            var value = $(this).val();
+            if (value && value != $(this).data('originalVariant')) {
```
Comment (aburgel) [inline src/popup.js:84]: only update if the variant changed

**Comment by HipsterBrown** [comment]
<< domain tafn

If we're introducing a `yarn.lock` file in this update, we should also update the build instructions to specify `yarn` to install dependencies instead of the current `npm install` instruction. I know `npm` version 7 can read `yarn.lock` files, however that just came out so we can't depend on it yet.

**Comment by samandmoore** [comment]
love this! thanks for figuring this out and fixing <3

sorry about all the jquery 😅

**Comment by aburgel** [comment]
bump!

**Comment by HipsterBrown** [comment]
DomainLGTM

**Pull request merged**